src/lib.rs

```diff
@@ -14,9 +14,9 @@ extern crate serde;
 #[cfg(feature = "with-serde")]
 extern crate serde_derive;
 
-#[cfg(unix)]
+#[cfg(linux)]
 mod unix;
```

Comment:
Should we  also rename the mod to linux?

Reply:
Yes, it would make sense for clarity. I didn't make that part of this pull request, because it would move a lot of files. But, it wouldn't be a disruptive change, since the 'unix' directory is flattened by the compile (for example, anything that depends on the Linux-specific src/unix/eventfd.rs uses "vmm_sys_util::eventfd" and not "vmm_sys_uti::unix::eventfd").

Reply:
Some modules in src/unix are not linux specific, such as file_trait, signal, tempdir, sock_msg_ctl etc. So should we change src/unix/mod.rs to use "#[cfg(linux)]" for feature control?

Reply:
That approach is also possible, here's what it looks like:

```
diff --git a/src/unix/mod.rs b/src/unix/mod.rs
index 8bf3202..3c4e36a 100644
--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -1,15 +1,26 @@
+#[cfg(linux)]
 #[macro_use]
 pub mod ioctl;
+#[cfg(linux)]
 pub mod aio;
+#[cfg(linux)]
 pub mod epoll;
+#[cfg(linux)]
 pub mod eventfd;
+#[cfg(linux)]
 pub mod fallocate;
 pub mod file_traits;
+#[cfg(linux)]
 pub mod poll;
+#[cfg(linux)]
 pub mod seek_hole;
+#[cfg(linux)]
 pub mod signal;
+#[cfg(linux)]
 pub mod sock_ctrl_msg;
 pub mod tempdir;
 pub mod terminal;
+#[cfg(linux)]
 pub mod timerfd;
+#[cfg(linux)]
 pub mod write_zeroes;
```
So, only file_traits, tempdir, and terminal successfully compile on non-Linux Unixes.

If you want to take this approach, I suggest moving all Linux-specific modules to src/linux, and keeping the ones intended to be compatible with all Unixes in src/unix. That clean separation makes the conditional compilation more obvious and maintainable, with a simple [#cfg(unix)] in src/lib.rs to include src/unix (for all Unix family OSs), and a simple [#cfg(linux)] in src/lib.rs to include src/linux (only for Linux).

Reply:
I'm ok to move to dedicated linux directory:) Need more feedbacks from other maintainers.
BTW, I expect sock_ctrl_msg should work on all unix too, maybe need to resolve some minor bugs.

Reply:
Nod, I'm happy to rework the patch to fit whatever approach the maintainers prefer.

The compile errors in sock_ctrl_msg are mostly type errors, specifically a usize type where the compiler expected either i32 or u32. It also uses unix/eventfd.rs, which also fails to compile on non-Linux, in that case because libc::eventfd doesn't exist, and neither do the constants libc::EFD_CLOEXEC, libc::EFD_NONBLOCK, and libc::EFD_SEMAPHORE.

So far, I haven't encountered use of sock_ctrl_msg in other rust-vmm crates, but eventfd is used extensively for device management in both Firecracker and Cloud Hypervisor, so it's likely I'll spend more time digging into that failure later.

Reply:
Looking into it a bit more, the OmniOS Illumos distro does have a Linux-style eventfd (added in the 2017 LTS release), but Rust's libc crate only conditionally compiles eventfd for Linux and Android.

Reply:
I would go forward as follows:
- add a linux module with the utilities that only build on Linux
- in the unix module only leave the utilities that build on Unix as well
- add GitHub issues for all Unix-compatible utilities that are now in the linux module, but could be "fixed" so that they properly work on Unix

This way @allisonrandal doesn't need to fix all the utilities. How does this sound?

Reply:
Great, so we could move forward fast:)

Reply:
I've updated the pull request to follow the approach @andreeaflorescu suggested